org.eclipse.scanning.points/src/org/eclipse/scanning/points/RepeatedPointIterator.java

```diff
@@ -30,29 +32,18 @@ public boolean hasNext() {
 		return count<model.getCount();
 	}
 
-	private static boolean countSleeps;
-	private static int     sleepCount;
-	   /**
-     * For testing we may count the sleeps of an interation
-     * @param b
-     */
-	public static void _setCountSleeps(boolean count) {
-		countSleeps = count;
-		sleepCount  = 0;
-	}
-
-	public static int _getSleepCount() {
-		return sleepCount;
-	}
-
 	@Override
 	public IPosition next() {
+		if (!hasNext()) {
+			throw new NoSuchElementException();
+		}
 
 		if (model.getSleep()>0) {
 			try {
 				Thread.sleep(model.getSleep());
 				if (countSleeps) sleepCount++;
 			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
 				throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)